etcdserver/etcdhttp/client_auth.go

```diff
@@ -208,20 +208,35 @@ func (sh *authHandler) forRole(w http.ResponseWriter, r *http.Request, role stri
 			return
 		}
 		if in.Role != role {
-			writeError(w, httptypes.NewHTTPError(401, "Role JSON name does not match the name in the URL"))
+			writeError(w, httptypes.NewHTTPError(http.StatusBadRequest, "Role JSON name does not match the name in the URL"))
 			return
 		}
-		newrole, created, err := sh.sec.CreateOrUpdateRole(in)
-		if err != nil {
-			writeError(w, err)
-			return
-		}
-		if created {
+
+		var out auth.Role
+
+		// create
+		if in.Grant.IsEmpty() && in.Revoke.IsEmpty() {
```

Comment:
What if the user passes the wrong thing (ie,  an uncreated user, with a Grant or a Revoke field set)